service/debugger/debugger.go

```diff
@@ -1381,14 +1429,33 @@ func (d *Debugger) Sources(filter string) ([]string, error) {
 	}
 
 	files := []string{}
-	for _, f := range d.target.BinInfo().Sources {
-		if regex.Match([]byte(f)) {
-			files = append(files, f)
+	for _, t := range d.target.Targets() {
```

Comment:
IMO returning a unified list for all processes provides the smoothest backwards compatibility for existing frontends. For example:
	
- getting a list of files and functions used by the program that the user can use to navigate the source code
- autocompletion for breakpoints
	
These are better served by the unified list rather than per-process lists. Dividing them per process would also need augmenting the API so if someone needs it we can add it later anyway.

Reply:
SGTM